classes/image_layer.py

```diff
@@ -86,8 +87,12 @@ def import_str(self, import_str):
         self.__import_str = import_str
 
     def add_package(self, package):
-        if package.name not in self.get_package_names():
-            self.__packages.append(package)
+        try:
+            assert isinstance(package, Package), 'Object type is {0}, it should be Package'.format(type(package))
```

Comment:
This looks great! One request: could you limit your line length to 80 characters (PEP8 standards)? You'll have to add a \ at the end I think.